pysyncobj/syncobj_admin.py

```diff
@@ -2,131 +2,63 @@
-                return False
-            self.__data = ['set_version', ver]
-            return True
-        else:
-            self.__result = 'invalid command'
-            return False
-
-
-    def __checkCorrectAddress(self, address):
-
-        try:
-            host, port = address.rsplit(':', 1)
-            port = int(port)
-            assert (port > 0 and port < 65536)
-            return True
-        except:
-            return False
-
-
-class Parser(object):
-    def __init__(self):
-        self.__parser = ArgumentParser()
-        self.__parser.add_argument('-conn', action='store', dest='connection', help='address to connect')
-        self.__parser.add_argument('-pass', action='store', dest='password', help='cluster\'s password')
-        self.__parser.add_argument('-status', action='store_true', help='send command \'status\'')
-        self.__parser.add_argument('-add', action='store', dest='add', help='send command \'add\'')
-        self.__parser.add_argument('-remove', action='store', dest='remove', help='send command \'remove\'')
-        self.__parser.add_argument('-set_version', action='store', dest='version', help='set cluster code version')
-
-    def parse(self, args):
-        return self.__parser.parse_args(args)
+from .utility import TcpUtility
+
+
+def checkCorrectAddress(address):
+    try:
+        host, port = address.rsplit(':', 1)
+        port = int(port)
+        assert (port > 0 and port < 65536)
+        return True
+    except:
+        return False
+
+
+def executeAdminCommand(args):
+    parser = ArgumentParser()
+    parser.add_argument('-conn', action='store', dest='connection', help='address to connect')
+    parser.add_argument('-pass', action='store', dest='password', help='cluster\'s password')
+    parser.add_argument('-status', action='store_true', help='send command \'status\'')
+    parser.add_argument('-add', action='store', dest='add', help='send command \'add\'')
+    parser.add_argument('-remove', action='store', dest='remove', help='send command \'remove\'')
+    parser.add_argument('-set_version', action='store', dest='version', type=int, help='set cluster code version')
+
+    data = parser.parse_args(args)
+    if not checkCorrectAddress(data.connection):
+        return 'invalid address to connect'
+
+    if data.status:
+        message = ['status']
+    elif data.add:
+        if not checkCorrectAddress(data.add):
+            return 'invalid address to command add'
+        message = ['add', data.add]
+    elif data.remove:
+        if not checkCorrectAddress(data.remove):
+            return 'invalid address to command remove'
+        message = ['remove', data.remove]
+    elif data.version is not None:
+        message = ['set_version', data.version]
+    else:
+        return 'invalid command'
+
+    util = TcpUtility(data.password)
+    result = util.executeCommand(data.connection, message)
+
+    if isinstance(result, str):
+        return result
```

Comment:
May be use only dict everywhere, and put "error" inside dict with error text for error handling?

Reply:
Well, the result could be anything (whatever the server returned).
I just "replicated" the [original code](https://github.com/bakwc/PySyncObj/pull/129/files#diff-db4d3a7131ef976c7af7c9b3af181bf959da3f8749e1e36829e33f5253fbb4feL38-L43).

Reply:
Ok, can merge it but looks not so cool. May be a separate issue for refactor?